### ISSUE TITLE
Sorting

### ISSUE THREAD

**Issue by exekutive** [josh-berry/tab-stash#194]
Sorry if this feature already exists, but I couldn't find it.

It would be really nice if there was a way to sort tabs inside a group alphabetically (and perhaps other criteria)

Keep up the good work.

<hr>
<em>Readers: If you are also interested in seeing this feature be developed, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the feature request, but "Me too!" comments are not counted by the voting system.</em>

**Comment by josh-berry** [comment]
Nope, I think you're the first person who's asked for it. :) Thanks for the feature request, I'll keep it in mind, and glad to hear Tab Stash is useful for you!

**Comment by josh-berry** [comment]
"Date" is unfortunately a bit harder, because Firefox Sync doesn't sync the date something was created. (From what I can tell, when something is first synced, it gets the current date, rather than the original creation date.) But I'll keep this in mind!

**Comment by mjbamford** [comment]
If it helps, there's the [Sort Bookmarks](https://addons.mozilla.org/en-US/firefox/addon/sort-bookmarks-webext/) add-on that will automatically keep bookmarks sorted by URL, Title or Date Added. It keeps the Tab Stash bookmarks in sorted order without side-affects. Quite exciting to see stashed tabs automatically sorted in Tab Stash's sidebar!

Btw, thanks for Tab Stash! It's marvelous!

**Comment by josh-berry** [comment]
@mjbamford You're quite welcome, and I'm glad you found a way to keep things sorted! Hopefully it's useful to others on this thread as well.

If you ever do run into a conflict, do let me know—I try hard to make sure Tab Stash plays nicely with other bookmark extensions, manual editing of bookmarks, etc. to the extent possible.

**Comment by ejm554** [comment]
**I came to this repo specifically to see if someone suggested a sorting feature. I'm glad that it is being considered!**

I'd like to request that the sorting be smart enough to sort by the domain names in the standard DNS form, i.e., beginning with the right-most (top-level) domain label, then the second-level domain label, and so on.

For example, doing a simple sort of URLs, e.g., alphabetically, might end up with a list that looks like this:

```
https://service.acer.com/chat/support/...
https://support.apple.com/en-us/HT201...
https://support.apple.com/en-us/HT203...
https://www.acer.com/ac/en/US/content...
https://www.acer.com/ac/en/US/search...
```
👆 Notice how the Acer sites are separated by the Apple sites?

If done using DNS form, it should look like this instead:

```
https://service.acer.com/chat/support/...
https://www.acer.com/ac/en/US/content...
https://www.acer.com/ac/en/US/search...
https://support.apple.com/en-us/HT201...
https://support.apple.com/en-us/HT203...
```

👆 Here the Acer sites are kept together.

I suggested this same idea for the FoxyTab extension, and it was eventually implemented. If you want to read the related GitHub Issues, they can be found [here](https://github.com/erosman/support/issues/407) and [here](https://github.com/erosman/support/issues/317).

**Comment by databill86** [comment]
I was just about to create a new issue to request this feature when I stumbled upon this existing issue. I'm hopeful that we'll be able to receive this feature soon. 
Thank you!

**Comment by Amarok73** [comment]
> I was just about to create a new issue to request this feature when I stumbled upon this existing issue. I'm hopeful that we'll be able to receive this feature soon.

So there's at least two of us :-)

**Comment by dec-mail** [comment]
> If it helps, there's the [Sort Bookmarks](https://addons.mozilla.org/en-US/firefox/addon/sort-bookmarks-webext/) add-on that will automatically keep bookmarks sorted by URL, Title or Date Added. It keeps the Tab Stash bookmarks in sorted order without side-affects. Quite exciting to see stashed tabs automatically sorted in Tab Stash's sidebar!
> 
> Btw, thanks for Tab Stash! It's marvelous!

i'm a bit confused.
TabStash is for Tabs. so how does a Bookmark Sort work?

**Comment by exekutive** [comment]
stashed tabs are stored as bookmarks. Look at your "other bookmarks" folder

**Comment by dec-mail** [comment]
hi there.
checked that and there  were only about a dozen tabs/bookmarks

![ShareX-2024-0501_180210](https://github.com/josh-berry/tab-stash/assets/149738734/6d736bcf-5eb2-4f77-a6de-7f08b2c10f81)

(which is a side issue to any actual sorting)

**Comment by exekutive** [comment]
Dunno 🤷‍♂️
That's where mine are.

**Comment by dec-mail** [comment]
Josh

Now i know app a little better, i don't have too much that i would enhance. :-)

So Sorting. i am not sure where we got with that.

So we need a Sorting function with a default Sort order, and some other Sorts.

The default Sort should be User set in settings/options; otherwise i suggest that a default Sort by (Alpha) URL DOMAIN only.

imo, Page Titles are not that useful as they are so random.

Other Sorts would be Date Saved ie latest on top (should prob be an option to reverse sort each option), and the full URL.

I can't actually think of any other useful Sorts.

Also maybe should have a Sort All (Tabs) and a Sort Within Group Only option

Group Sorts should just be Alpha or as manually sorted.
